### PR TITLE
Fix: 182 internal server error bug when 400+ requests

### LINKED ISSUE #182: Internal server error bug when 400+ requests
**Context**

- Log: (psycopg2.OperationalError) connection to server at "34.27.217.4", port 5432 failed: FATAL:  remaining connection slots are reserved for non-replication superuser connections
- Connection metrics from GCP at the time of the issue 
Screenshot 2023-11-25 at 2 19 57 PM

**Possible Solutions**
- Load API test

### LINKED ISSUE #219: Use a global session for GCP database connection
### Describe the problem

We need to use a global session in api/src/databse/database.py like what we have in functions-python/helpers/database.py. 

### Proposed solution

We need a global session per container, when container reloads, we reuse the global session. For our application, what we do is database selections. Creating and closing each session is not maintainable.

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**

Closes #182 
Closes #219 
**Expected behavior:** 

I need this PR to be merged coz I need to do load test on QA. The load test on DEV looks good - Active connection is around 4 on GCP when Ramping to 2000 users at a rate of 100.00 per second.

image

**Testing tips:**

Provide tips, procedures and sample files on how to test the feature.
Testers are invited to follow the tips AND to try anything they deem relevant outside the bounds of the testing tips. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (davidgamez) [inline .vscode/launch.json:1]: Please local file from the repository.

File: api/src/database/database.py [hunk 2]
```diff
@@ -170,13 +192,16 @@ def merge(
...
             if auto_commit:
-                self.session.commit()
+                global_session.commit()
             return True
         except Exception as e:
             self.logger.error(f"Merge query failed with exception: \n{e}")
             return False
+        # finally:
+        #     if not update_session:
+        #         self.close_session()
```
Comment (cka-y) [inline api/src/database/database.py:204]: Does this need to stay commented?

File: load-test/gtfs_user_test.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from locust import HttpUser, TaskSet, task, between
```
Comment (cka-y) [inline load-test/gtfs_user_test.py:1]: Great addition! Could you provide some documentation on how to setup/run the load tests locally?
Suggestion: Create or add to an existing GH action the load tests using the cli

File: api/src/database/database.py [hunk 4]
```diff
@@ -50,34 +51,52 @@ def is_connected(self):
...
                 self.engine = create_engine(self.SQLALCHEMY_DATABASE_URL, echo=True)
-                self.logger.debug("Database connected.")
-            if self.session is not None and self.session.is_active:
-                self.session.close()
-            self.session = Session(self.engine, autoflush=False)
-        except Exception as e:
-            self.logger.error(f"Database new session creation failed with exception: \n {e}")
-        return self.is_connected()
+                global_session = sessionmaker(bind=self.engine)()
+                self.session = global_session
```
Comment (davidgamez) [inline api/src/database/database.py:85]: I think the _self.session_ is only used in this assignment as it's replaced by accessing the _ global_session_ directly. If this is true, let's remove this assignment.

**Review by davidgamez — APPROVED** [review]

**Pull request merged**